www/js/main.js

```diff
@@ -24,9 +36,76 @@ define([
 		} else {
 			throw Error("something went wrong");
 		}
+	
+	//The following code follows sachin code after the model has been rendered according to query parameters
+	ready(function(){
+		//remove the loading division, now that the problem is being loaded
+		var loading = document.getElementById('loadingOverlay');
+		loading.style.display = "none";
+
+		var dm = new drawModel(_model.active);
+
+		var menuButtons=[];
+		menuButtons.push("createQuantityNodeButton");
+		menuButtons.push("createEquationNodeButton");
+
+		array.forEach(menuButtons, function(button){
+			//setting display for each menu button
+			style.set(registry.byId(button).domNode, "display", "inline");
+
+			/*
+		 	* This is a work-around for getting a button to work inside a MenuBar.
+		 	* Otherwise, there is a superfluous error message.
+		 	*/	
+		 	registry.byId(button)._setSelected = function(arg){
+				console.log(button+" _setSelected called with ", arg);
+			}
+			}, this);
+
+		//check if the current UI situation permits loading of the createNodeButton
+		//for now since we are not into UI parameters like in dragoon yet
+		//I enable the button without any check
+		var createQNodeButton = registry.byId("createQuantityNodeButton");
+		createQNodeButton.setDisabled(false);
+
+		var createEqNodeButton = registry.byId("createEquationNodeButton");
+		createEqNodeButton.setDisabled(false);
 
-		ready(function(){
-			var dm = new drawModel(_model.active);
+		//create a controller object
+
+		//create new model object
+		var givenModel = "";
+	
+		//create new ui configuration object based on current mode ( and activity)
+		var ui_config = "";
+
+		//For now using empty givenModel and ui_config 
+
+		var controllerObject = new controlAuthor(query.m, givenModel, ui_config);
```

Comment:
Its AOK from what I think. I dont know why it was showing me an old commit where it was obj["authored"]. I remembered and rechecked it. Still need to get the hang of this thing.